api_app/visualizers_manager/visualizers/passive_dns/analyzer_extractor.py

```diff
@@ -0,0 +1,171 @@
+"""In this module there are functions to extract the data required by
+Passive DNS visualizer from the analyzers reports.
+"""
+
+import datetime
+import logging
+from typing import List
+
+from django.db.models import QuerySet
+
+from api_app.analyzers_manager.models import AnalyzerReport
+from api_app.analyzers_manager.observable_analyzers.circl_pdns import CIRCL_PDNS
+from api_app.analyzers_manager.observable_analyzers.dnsdb import DNSdb
+from api_app.analyzers_manager.observable_analyzers.mnemonic_pdns import (
+    MnemonicPassiveDNS,
+)
+from api_app.analyzers_manager.observable_analyzers.otx import OTX
+from api_app.analyzers_manager.observable_analyzers.robtex import Robtex
+from api_app.analyzers_manager.observable_analyzers.threatminer import Threatminer
+from api_app.analyzers_manager.observable_analyzers.validin import Validin
+from api_app.models import Job, PythonModule
+
+logger = logging.getLogger(__name__)
+
+
+def _extract_analyzer(
+    analyzer_reports: QuerySet, module: PythonModule, job: Job
+) -> AnalyzerReport:
+    try:
+        analyzer_report = analyzer_reports.get(config__python_module=module)
+        printable_analyzer_name = analyzer_report.config.name.replace("_", " ")
+        logger.debug(f"{printable_analyzer_name=}")
+    except AnalyzerReport.DoesNotExist:
+        logger.warning(f"job: {job.id}, {module} analyzer report doesn't exist")
+        return None
+    return analyzer_report
+
+
+def extract_otxquery_reports(analyzer_reports: QuerySet, job: Job) -> List:
+    otx_analyzer = _extract_analyzer(analyzer_reports, OTX.python_module, job)
+    if otx_analyzer:
+        otx_reports = otx_analyzer.report.get("passive_dns", [])
+        for report in otx_reports:
+            report["last_view"] = report.pop("last").split("T")[0]
+            report["first_view"] = report.pop("first").split("T")[0]
+            report["rrtype"] = report.pop("record_type")
+            report["rdata"] = report.pop("address")
+            report["rrname"] = report.pop("hostname")
+            report["source"] = otx_analyzer.config.name.replace("_", " ")
+            report["source_description"] = otx_analyzer.config.description
```

Comment:
I think a dataclass is better than a dict: In this case the fields are not too much (6) and I think we are not going to modify them, but a dataclass define the fields and it's easier to know what fields are available.